java/com/google/location/lbs/activity/audioset/dataprocessing/CsvParser.java

```diff
@@ -0,0 +1,244 @@
+import java.io.File;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Arrays;
+
+/**
+ * CSV Class
+ *
+ * <p>CSV is a class for the reading of a dataset in a csv format.
+ *
+ * @author Carver Forbes
+ * @version 0.1
+ */
+public class CsvParser {
+
+  private File file;
+  private ArrayList<String> ytidList;
+  private HashMap<String, IncompleteExample.Example> idToProtoMap;
+  private HashMap<String, String> idToUrlMap;
+
+  private final boolean debug = true;
+
+  /**
+   * CSV constructor
+   *
+   * @param pathname the pathname for the csv file to be read
+   * @throws IllegalArgumentException if the pathname is incorrect
+   */
+  public CsvParser(String pathname) {
+    file = new File(pathname);
+    if (!file.exists()) {
+      throw new IllegalArgumentException(String.format("This file does not exist: %s", pathname));
+    }
+    ytidList = new ArrayList<>();
+    idToProtoMap = new HashMap<>();
+    idToUrlMap = new HashMap<>();
+  }
+
+  /** Tester Function to print 5 YouTube Links */
+  public void printYouTubeLinks() {
+    // prints youtube links
+    Scanner scan = null;
+    try {
+      scan = new Scanner(file);
+      String line = scan.nextLine();
+      while (line.charAt(0) == '#') {
+        line = scan.nextLine();
+      }
+      // while (scan.hasNext()) {
+      //   String ytid = line.split(",")[0];
+      //   String url = "https://youtube.com/watch?v=" + ytid;
+      //   System.out.println(url);
+      //   line = scan.nextLine();
+      // }
+      for (int i = 0; i < 5; i++) {
+        String ytid = line.split(",")[0];
+        String url = "https://youtube.com/watch?v=" + ytid;
+        System.out.println(url);
+        line = scan.nextLine();
+      }
+    } catch (IOException io) {
+      io.printStackTrace();
+    } finally {
+      if (scan != null) {
+        scan.close();
+      }
+    }
+  }
+
+  /** Extracts metadata from dataset.csv file */
+  public void csvToProtoMap() {
+    if (debug) System.out.println("in the csvToProtoMap method");
+    Scanner scan = null;
```

Comment:
if it's possible, i'd avoid writing my own csv parser -- there are some you can use?

Reply:
Here's one from [`ApacheCommons`](http://commons.apache.org/proper/commons-csv/apidocs/org/apache/commons/csv/CSVParser.html)

Reply:
Ok. Thanks. So even in simple use case scenarios existing csv parser libraries are preferred?